### ISSUE TITLE
Handle generic classes without warnings

### ISSUE THREAD

**Issue by ChristianBecker** [JakeWharton/butterknife#651]
After switching from `.compilesWithoutError()` to `.compilesWithoutWarnings()` (see #650) we see a couple of tests fail with the following message:
```
found raw type: test.Test
  missing type arguments for generic class test.Test<T>
```

The outline of the generated source is printed below:
```
    // source
    class Test<T extends TextView> extends Activity { … }

    // binderSource
    public final class Test_ViewBinder implements ViewBinder<Test> { … }

    // bindingSource
    public class Test_ViewBinding<T extends Test> implements Unbinder { … }
```

How would you change the generated sources to get rid of this warning?

**Comment by JakeWharton** [comment]
I suspect this gets complicated if you're using `T` as a field type that's being bound...

**Comment by Wavesonics** [comment]
Handling these edge cases maybe not a rabbit hole that's worth exploring.

It would still be great though if we could get on with our warningless lives :p 

Maybe the generated classes could just suppress their potential warnings at the class level:
@SuppressWarnings("all")

**Comment by Wavesonics** [comment]
@JakeWharton I'd be happy to make a pull request adding a @SuppressWarnings({"rawtypes"}) annotation to the generated classes if that's something you'd want to accept?

Currently this is the last source of warnings in my project ;)

**Comment by JakeWharton** [comment]
I haven't seen what an attempt to support this properly looked like yet in
order to settle for throwing up my hands in defeat.

**Comment by Wavesonics** [comment]
Fair enough, if I have some time in the near future I'll take a crack at trying to propagate the generics through into the codegen.

**Comment by JakeWharton** [comment]
Unless the generic is used in the actual bindings (which I'm not sure we allow...), we should be able to just use wildcards.
